src/components/PopupWithForm.js

```diff
@@ -3,21 +3,31 @@ import Popup from "./Popup.js";
 export default class PopupWithForm extends Popup {
   #submitHandler;
   #form;
+  #submitButton;
 
   constructor(popupSelector, submitHandler) {
     super(popupSelector);
     this.#submitHandler = submitHandler;
     this.#form = this.modalElement.querySelector(".modal__form");
+    this.#submitButton = this.#form.querySelector(".modal__save-button");
   }
 
   #getInputValues() {
-    this.inputInfo = {};
+    const inputInfo = {};
     const inputs = Array.from(this.modalElement.querySelectorAll(".modal__input"));
     inputs.forEach(input => {
-      this.inputInfo[input.placeholder] = input.value;
+      inputInfo[input.name] = input.value;
     })
 
-    return this.inputInfo;
+    return inputInfo;
+  }
+
```

Comment:
**Might be better**


You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`